File: countGoodNumbers.js
var countGoodNumbers = function (n) {
  const MOD = 1_000_000_007n;
  let result = n % 2 === 0 ? 1n : 5n;
  let x = 20n;

  for (let i = BigInt(Math.floor(n / 2)); i > 0n; i /= 2n) {
    if (i % 2n !== 0n) {
      result = (result * x) % MOD;
    }
    x = (x * x) % MOD;
  }

  return Number(result);
};

console.log(countGoodNumbers(8)); // 160000

File: countGoodNumbers.py
def countGoodNumbers(n: int) -> int:
        MOD = 10**9 + 7
        even_count = n // 2
        odd_count = (n + 1) // 2

        return (pow(5, odd_count, MOD) * pow(4, even_count, MOD)) % MOD

print(countGoodNumbers(3)) # 100
print(countGoodNumbers(5)) # 2000